plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java

```diff
@@ -102,13 +129,53 @@ public DockerConnectorConfiguration(URI dockerDaemonUri,
         this.networkFinder = networkFinder;
     }
 
-    public static String getExpectedLocalHost() {
-        return SystemInfo.isLinux() ? "localhost" : dockerDaemonUri().getHost();
+    public static boolean isUnixSocketUri(URI uri) {
```

Comment:
What if `uri` is null ?
Don't you think that it would be better to define this sort of contract here?